docs/guides/patterns/knativetrigger.md

```diff
@@ -0,0 +1,179 @@
+
+If on the other your Knative service expects a CloudEvent, or you'd like an example of how to build one from the ground up, then take a look at [this post](https://www.triggermesh.com/blog/build-a-custom-triggermesh-component-with-fastapi) that shows how to build a CloudEvents Knative service using Python FastAPI and route CloudEvents to it with TriggerMesh. 
+
+The solution we’ll build in this guide is illustrated in the schema below.
+
+![](../../assets/images/knative-trigger/schema.png)
+
+## Setup
+
+If you haven’t already done so, you’ll need to install TriggerMesh and Knative Serving on your cluster. I’m using [Minikube](https://minikube.sigs.k8s.io/docs/start/) to create a local cluster and I [followed the guide](../../installation/kubernetes-yaml.md) for installing TriggerMesh and Knative Serving onto the cluster. 
+
+I’m going to use a simple containerised http service that I [found here](https://hub.docker.com/r/strm/helloworld-http/) to run as a Knative service. Feel free to replace this with your own of course. 
+
+Define the Knative service in a manifest called knative-service.yaml:‍‍
+
+```yaml
+apiVersion: serving.knative.dev/v1
+kind: Service
+metadata:
+  name: hello-http
+spec:
+  template:
+    spec:
+      containers:
+        - image: strm/helloworld-http
+          ports:
+          - containerPort: 80
+```
+
+Apply this to the cluster:‍‍
+
+```sh
+kubectl apply -f knative-service.yaml
+```
+
+Check that the Knative service is running with the following command:‍‍
+
+```sh
+kubectl get ksvc hello-http 
+NAME         URL                                          LATESTCREATED     LATESTREADY       READY   REASON
+hello-http   http://hello-http.default.svc.cluster.local  hello-http-00002  hello-http-00002  True  
+```
+
+Now deploy a TriggerMesh broker to the cluster. The broker decouples event producers and consumers, and provides triggers that send events to consumers if they match a given filter. Create the following broker.yaml:‍‍
+
+```yaml
+apiVersion: eventing.triggermesh.io/v1alpha1
+kind: MemoryBroker
+metadata:
+  name: broker
+```
+
+And apply it to the cluster:‍‍
+
+```sh
+kubectl apply -f broker.yaml
+```
+
+Check that the broker is available:‍‍
+
+```sh
+kubectl get memorybrokers.eventing.triggermesh.io 
+NAME     URL                                                 AGE     READY   REASON
+broker   http://broker-mb-broker.default.svc.cluster.local   3h38m   True    
+```
+
+Now create a Webhook source, in a file called webhook-source.yaml, that we'll use as an event producer. It will let us easily send events of type `my.event.type` to the broker over HTTP. Later, you can replace it or add additional event sources like Kafka, AWS SQS, Google Pub/Sub, etc… ‍‍
```

Comment:
```suggestion
Now create a Webhook source, in a file called `webhook-source.yaml`, that we'll use as an event producer. It will let us easily send events of type `my.event.type` to the broker over HTTP. Later, you can replace it or add additional event sources like Kafka, AWS SQS, Google Pub/Sub, etc… ‍‍
```